tests/test_pipeline/test_schedule/test_zerobubble_pp.py

```diff
@@ -0,0 +1,734 @@
+from copy import deepcopy
+from typing import Tuple
+
+import pytest
+import torch
+import torch.distributed as dist
+import torch.nn as nn
+from torch.testing import assert_close
+
+import colossalai
+from colossalai.cluster import ProcessGroupMesh
+from colossalai.interface import OptimizerWrapper
+from colossalai.logging import disable_existing_loggers
+from colossalai.pipeline.schedule.v_schedule import PipelineGraph, ScheduledNode
+from colossalai.pipeline.schedule.zero_bubble_pp import ZeroBubbleVPipeScheduler
+from colossalai.pipeline.stage_manager import PipelineStageManager
+from colossalai.testing import parameterize, rerun_if_address_is_in_use, spawn
+from tests.kit.model_zoo import model_zoo
+
+
+class MlpModel(nn.Module):
+    def __init__(self, in_dim, out_dim, num_layers):
+        super().__init__()
+        self.layers = nn.ModuleList([nn.Linear(in_dim, out_dim, bias=None) for _ in range(num_layers)])
+
+    def forward(self, x):
+        for layer in self.layers:
+            x = layer(x)
+        return x
+
+
+def get_model_numel(model: torch.nn.Module) -> Tuple[int, int]:
+    num_params = 0
+    num_params_trainable = 0
+    for p in model.parameters():
+        num_params += p.numel()
+        if p.requires_grad:
+            num_params_trainable += p.numel()
+    return num_params, num_params_trainable
+
+
+# 1) Test manual v_schedule with multiple microbatch
+@parameterize(
+    "test_config",
+    [
+        {
+            "batch_size": 4,
+            "tp_size": 1,
+            "pp_size": 4,
+            "num_microbatches": 4,
```

Comment:
increase this value and batch size to 8

Reply:
> increase this value and batch size to 8

Update in https://github.com/hpcaitech/ColossalAI/pull/6034/commits/e6e1a97a6d2d69fc8cd2907883e0627a61e6f372;